src/components/common/Menu.svelte

```diff
@@ -54,21 +56,23 @@
       return;
     }
 
+    const containerRect = containerEl.getBoundingClientRect();
     const activatorRect = activator.getBoundingClientRect();
     const offsetY = activator.clientHeight + 5;
-    if (activatorRect.bottom + listDiv.clientHeight > windowInnerHeight) {
+    if (activatorRect.top - listDiv.clientHeight > containerRect.top) {
       offsetYStyle = `bottom: ${offsetY}px;`;
     } else {
       offsetYStyle = `top: ${offsetY}px;`;
     }
 
-    if (activatorRect.right + listDiv.clientWidth > windowInnerWidth) {
+    if (activatorRect.right + listDiv.clientWidth > containerRect.right) {
       offsetX = 'right-0';
     } else {
       offsetX = 'left-0';
     }
   };
 
+  $: containerEl = document.querySelector(containerQuery) ?? document.body;
```

Comment:
Might want to move this to right before `containerEl.getBoundingClientRect()`. The element is yeeted whenever the settings page is opened/closed, so menus on old messages will break while new ones are fine:

https://user-images.githubusercontent.com/20059164/184477711-7a3cac6e-2ce2-4182-822f-1323ef39888c.mp4

Reply:
Fixed in [deb6df7](https://github.com/LiveTL/LiveTL/pull/426/commits/deb6df70e4ea9b9eb5d22311ccaea5644462ccef)